chia/full_node/block_store.py

```diff
@@ -151,22 +170,22 @@ async def add_full_block(
             )
 
             await self.db.execute(
-                "INSERT OR REPLACE INTO full_blocks VALUES(?, ?, ?, ?, ?, ?, ?, ?)",
+                "INSERT OR IGNORE INTO full_blocks VALUES(?, ?, ?, ?, ?, ?, ?, ?)",
                 (
                     header_hash,
                     block.prev_header_hash,
                     block.height,
                     ses,
                     int(block.is_fully_compactified()),
-                    in_main_chain,  # in_main_chain
+                    False,  # in_main_chain
```

Comment:
looks like we never use this now, maybe a comment if cant remove the column

Reply:
This field is new in the v2 schema and I expect us to rely more on it, as it can simplify a fair number of lookups based on height.

we use the `in_main_chain` when looking for blocks to compactify, and there are a fair number of places where we currently rely on the height-to-hash map in memory because we didn't used to have this field. Some of those places can be relaxed to not need the height-to-hash map in memory.

Reply:
very cool, i didn't realize we keep this updated in reconsider_peek